packages/extension/src/companion/App.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { ReactElement, useState } from 'react';
+import { QueryClient, QueryClientProvider } from 'react-query';
+import { browser } from 'webextension-polyfill-ts';
+import { Boot } from '@dailydotdev/shared/src/lib/boot';
+import FeaturesContext from '@dailydotdev/shared/src/contexts/FeaturesContext';
+import { AuthContextProvider } from '@dailydotdev/shared/src/contexts/AuthContext';
+import { SettingsContextProvider } from '@dailydotdev/shared/src/contexts/SettingsContext';
+import { AlertContextProvider } from '@dailydotdev/shared/src/contexts/AlertContext';
+import { AnalyticsContextProvider } from '@dailydotdev/shared/src/contexts/AnalyticsContext';
+import { RouterContext } from 'next/dist/shared/lib/router-context';
+import Companion from './Companion';
+import CustomRouter from '../lib/CustomRouter';
+import { companionFetch } from './companionFetch';
+
+const queryClient = new QueryClient();
+const router = new CustomRouter();
+
+export type CompanionData = { url: string; deviceId: string } & Pick<
+  Boot,
+  'postData' | 'settings' | 'flags' | 'alerts' | 'user' | 'visit'
+>;
+export default function App({
+  deviceId,
+  url,
+  postData,
+  settings,
+  flags,
+  user,
+  alerts,
+  visit,
+}: CompanionData): ReactElement {
+  const [isOptOutCompanion, setIsOptOutCompanion] = useState<boolean>(
+    settings?.optOutCompanion,
+  );
+  if (isOptOutCompanion) {
+    return <></>;
+  }
+
+  return (
+    <div>
+      <style>
+        @import &quot;chrome-extension://{browser.runtime.id}
+        /css/companion.css&quot;;
+      </style>
+      <RouterContext.Provider value={router}>
+        <QueryClientProvider client={queryClient}>
+          <FeaturesContext.Provider value={{ flags }}>
```

Comment:
Usually, it's better to have `{{ flags }}` memoized. Otherwise it creates a new object on every render